base/complex.jl

```diff
@@ -290,6 +290,13 @@ inv(z::Complex{<:Integer}) = inv(float(z))
 *(z::Complex, w::Complex) = Complex(real(z) * real(w) - imag(z) * imag(w),
                                     real(z) * imag(w) + imag(z) * real(w))
 
+function *(z::Complex{Union{BigInt,BigFloat}}, w::Complex{Union{BigInt,BigFloat}})
```

Comment:
```suggestion
function *(z::Complex{<:Union{BigInt,BigFloat}}, w::Complex{<:Union{BigInt,BigFloat}})
```
because
```julia
julia> Complex(big(2)) isa Complex{Union{BigInt,BigFloat}}
false

julia> Complex(big(2)) isa Complex{<:Union{BigInt,BigFloat}}
true
```